Source/WebCore/platform/graphics/mac/DisplayConfigurationMonitor.h

```diff
@@ -44,13 +46,23 @@ class DisplayConfigurationMonitor {
         virtual ~Client();
         virtual void displayWasReconfigured() = 0;
     };
-    void addClient(Client&);
+    void addClient(Client&, FunctionDispatcher* = nullptr);
     void removeClient(Client&);
 
     WEBCORE_EXPORT void dispatchDisplayWasReconfigured(CGDisplayChangeSummaryFlags);
 private:
     DisplayConfigurationMonitor();
-    Vector<Client*> m_clients;
+    Lock m_lock;
+    struct ClientEntry {
+        Client* m_client;
```

Comment:
Should this use the fancy new thread safe weak pointer thing?

Reply:
That's for things that are both refcounted and use weakptr. I'm not sure it helps us much here, since we're not refcounting Client objects.